store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/FederatedGraphStorage.java

```diff
@@ -598,23 +609,68 @@ private boolean changeGraphId(final String graphId, final String newGraphId, fin
                 }
             }
 
-            final GraphConfig configWithNewGraphId = new GraphConfig.Builder()
-                    .json(new GraphSerialisable.Builder().graph(graphToMove).build().getConfig())
-                    .graphId(newGraphId)
-                    .build();
+            //Update Tables
+            String storeClass = graphToMove.getStoreProperties().getStoreClass();
+            if (nonNull(storeClass) && storeClass.startsWith(AccumuloStore.class.getPackage().getName())) {
+                /*
+                 * uk.gov.gchq.gaffer.accumulostore.[AccumuloStore, SingleUseAccumuloStore,
+                 * SingleUseMockAccumuloStore, MockAccumuloStore, MiniAccumuloStore]
+                 */
+                try {
+                    AccumuloProperties tmpAccumuloProps = (AccumuloProperties) graphToMove.getStoreProperties();
+                    Connector connection = TableUtils.getConnector(tmpAccumuloProps.getInstance(),
+                            tmpAccumuloProps.getZookeepers(),
+                            tmpAccumuloProps.getUser(),
+                            tmpAccumuloProps.getPassword());
+
+                    if (connection.tableOperations().exists(graphId)) {
+                        connection.tableOperations().offline(graphId);
+                        connection.tableOperations().clone(graphId, newGraphId, true, null, null);
```

Comment:
Rather than cloning and deleting the table which is costly for large amounts of data, can't you do something like `connection.tableOperations().rename(graphId, newGraphId);`?

Reply:
Accumulo is smarter that that, the data doesn't copy.

Reply:
Fair enough, but since this is a rename operation, would it not just make sense to rename it instead of cloning and deleting?

Reply:
https://accumulo.apache.org/1.5/apidocs/org/apache/accumulo/core/client/admin/TableOperations.html
I was able to swap it with rename and it worked just fine

Reply:
what is this witchcraft. I was unable to find a rename.
or am I thinking of the command line Accumulo api.
maddening.